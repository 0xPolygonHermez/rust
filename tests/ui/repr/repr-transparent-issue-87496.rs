// Regression test for the ICE described in #87496.

// check-pass

#[repr(transparent)]
struct TransparentCustomZst(());
extern "C" {
    fn good17(p: TransparentCustomZst);
    //~^ WARNING: `extern` block uses type `()`, which is not FFI-safe
}

fn main() {}
