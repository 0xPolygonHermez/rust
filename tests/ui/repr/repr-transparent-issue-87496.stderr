warning: `extern` block uses type `()`, which is not FFI-safe
  --> $DIR/repr-transparent-issue-87496.rs:8:18
   |
LL |     fn good17(p: TransparentCustomZst);
   |                  ^^^^^^^^^^^^^^^^^^^^ not FFI-safe
   |
   = help: consider using a struct instead
   = note: tuples have unspecified layout
   = note: `#[warn(improper_ctypes)]` on by default

warning: 1 warning emitted

